typespec-tests/tsp/enum.tsp

```diff
@@ -19,10 +19,11 @@ enum Color {
   Green,
 }
 
-enum ColorModel {
-  Red,
-  Blue,
-  Green,
+union ColorModel {
+  string,
+  Red: "Red",
+  Blue: "Blue",
```

Comment:
why is this key value pairs? i.e. `Red: "Red"`, but others are not?

Reply:
There is a line ref to that UnionVariant
```
  colorModelValue: ColorModel.Blue;
```

so it needs a name.

Reply:
I see.